diesel/src/row.rs

```diff
@@ -148,6 +148,15 @@ where
     }
 }
 
+/// A row that can be turned into an owned version
+pub trait IntoOwnedRow<'a, DB: Backend> {
```

Comment:
I would like to gate that behind the `i-implement-a-third-party-backend-and-opt-into-breaking-changes` feature, as it's not useful for anyone outside of diesel-async/other crates doing really low level stuff.

Also maybe just make this a super trait of `Row`, so

```suggestion
pub trait IntoOwnedRow<'a, DB: Backend>: Row<'a, DB> {
```